internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -0,0 +1,1579 @@
+		return
+	}
+
+	// Read the response into the state
+	var state oauthClientResourceModel
+
+	diags = readOauthClientResponse(ctx, oauthClientResponse, &plan, &state)
+	resp.Diagnostics.Append(diags...)
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *oauthClientResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state oauthClientResourceModel
+
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	apiReadOauthClient, httpResp, err := r.apiClient.OauthClientsAPI.GetOauthClientById(config.ProviderBasicAuthContext(ctx, r.providerConfig), state.ClientId.ValueString()).Execute()
+
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
+			config.ReportHttpErrorAsWarning(ctx, &resp.Diagnostics, "An error occurred while getting the OauthClient", err, httpResp)
+			resp.State.RemoveResource(ctx)
+		} else {
+			config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while getting the  OauthClient", err, httpResp)
+		}
+	}
+
+	// Read the response into the state
+	readOauthClientResponse(ctx, apiReadOauthClient, &state, &state)
+
+	// Set refreshed state
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *oauthClientResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	var plan oauthClientResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	updateOauthClient := r.apiClient.OauthClientsAPI.UpdateOauthClient(config.ProviderBasicAuthContext(ctx, r.providerConfig), plan.ClientId.ValueString())
+	createUpdateRequest := client.NewClient(plan.ClientId.ValueString(), grantTypes(plan.GrantTypes), plan.Name.ValueString())
+	err := addOptionalOauthClientFields(ctx, createUpdateRequest, plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to add optional properties to add request for OauthClient", err.Error())
+		return
+	}
+
+	updateOauthClient = updateOauthClient.Body(*createUpdateRequest)
+	updateOauthClientResponse, httpResp, err := r.apiClient.OauthClientsAPI.UpdateOauthClientExecute(updateOauthClient)
+	if err != nil && (httpResp == nil || httpResp.StatusCode != 404) {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while updating OauthClient", err, httpResp)
+		return
+	}
+
+	// Read the response
+	var state oauthClientResourceModel
+	diags = readOauthClientResponse(ctx, updateOauthClientResponse, &plan, &state)
+	resp.Diagnostics.Append(diags...)
+
+	// Update computed values
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+}
+
+// This config object is edit-only, so Terraform can't delete it.
```

Comment:
Remove this comment

Reply:
Found this is a few other resources as well. This will be included in the next commit

Reply:
updated